src/transformers/models/vocos/modeling_vocos.py

```diff
@@ -0,0 +1,368 @@
+            Reconstructed audio waveform.
+    """
+
+    audio: torch.FloatTensor
+
+
+class VocosAdaptiveLayerNorm(nn.Module):
+    """
+    Weight and bias parameters come from a lookup table based on the target bandwidth.
+    """
+
+    def __init__(self, config: VocosConfig):
+        super().__init__()
+        self.eps = config.layer_norm_eps
+        self.hidden_size = config.hidden_size
+        adanorm_num_embeddings = len(config.bandwidths)
+        self.weight = nn.Parameter(torch.ones(adanorm_num_embeddings, config.hidden_size))
+        self.bias = nn.Parameter(torch.zeros(adanorm_num_embeddings, config.hidden_size))
+
+    def forward(self, hidden_states: torch.Tensor, cond_embedding_id: torch.LongTensor):
+        hidden_states = F.layer_norm(hidden_states, (self.hidden_size,), weight=None, bias=None, eps=self.eps)
+        return hidden_states * self.weight[cond_embedding_id].unsqueeze(0) + self.bias[cond_embedding_id].unsqueeze(0)
+
+
+class VocosConvNeXtBlock(nn.Module):
+    """ConvNeXt block adapted for 1D convolutions in the Vocos architecture."""
+
+    def __init__(self, config: VocosConfig):
+        super().__init__()
+        self.dwconv = nn.Conv1d(
+            config.hidden_size,
+            config.hidden_size,
+            kernel_size=config.kernel_size,
+            padding=config.padding,
+            groups=config.hidden_size,
+        )
+        if config.use_adaptive_norm:
+            self.norm = VocosAdaptiveLayerNorm(config)
+        else:
+            self.norm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.pwconv1 = nn.Linear(config.hidden_size, config.intermediate_size)
+        self.act = nn.GELU()
+        self.pwconv2 = nn.Linear(config.intermediate_size, config.hidden_size)
+        if config.layer_scale_init_value > 0:
+            self.layer_scale_parameter = nn.Parameter(
+                config.layer_scale_init_value * torch.ones(config.hidden_size), requires_grad=True
+            )
+        else:
+            self.layer_scale_parameter = None
+
+    def forward(self, hidden_states: torch.Tensor, bandwidth_id: Optional[torch.LongTensor] = None) -> torch.Tensor:
+        residual = hidden_states
+        hidden_states = self.dwconv(hidden_states)
+        hidden_states = hidden_states.transpose(1, 2)
+        if isinstance(self.norm, VocosAdaptiveLayerNorm):
+            hidden_states = self.norm(hidden_states, cond_embedding_id=bandwidth_id)
+        else:
+            hidden_states = self.norm(hidden_states)
+        hidden_states = self.pwconv1(hidden_states)
+        hidden_states = self.act(hidden_states)
+        hidden_states = self.pwconv2(hidden_states)
+        if self.layer_scale_parameter is not None:
+            hidden_states = self.layer_scale_parameter.to(hidden_states.device) * hidden_states
+        hidden_states = hidden_states.transpose(1, 2)
+        hidden_states = residual.to(hidden_states.device) + hidden_states
```

Comment:
Needed to `.to(device)` for  `device_map="auto"` to work when loading form pretrained (VocosModel.from_pretrained(model_id, device_map="auto"))